src/mpl/src/sock/mpl_sockaddr.c

```diff
@@ -0,0 +1,283 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2018 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+/* Rationale:
+    *   MPL wrap for handling IPv4 and IPv6.
+    *
+    *   Applications: pm, pmi, ch3.
+    *   ch4 supports tcp sockets indirectly through ucx and ofi.
+    *
+    * Design consideration:
+    *   Either IPv4 or IPv6, globally set as defalt or with command line option, to simplify logic.
+    *   TCP only, no UDP or unix domain sockets.
+    *
+    *   Application use struct sockaddr_storage exclusively.
+    *       MPL_get_sockaddr for hostname
+    *       MPL_get_sockaddr_iface for network interface
+    *       MPL_get_sockaddr_direct for listening socket on ANY or LOOPBACK
+    *
+    *   Simplified MPL_connect and MPL_listen interface.
+    *       Both have a port parameter.
+    *       MPL_listen combines bind with listen.
+*/
+
+#include "mplconfig.h"
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <string.h>
+#include <ifaddrs.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include "mpl_sockaddr.h"
+
+static int is_localhost(struct sockaddr *p_addr);
+
+static int af_type = AF_INET;
+static int _use_loopback = 0;
+static int _max_conn = SOMAXCONN;
+
+int MPL_get_sockaddr(const char *s_hostname, struct sockaddr_storage *p_addr)
+{
+    struct addrinfo ai_hint;
+    struct addrinfo *ai_list;
+    int ret;
+    memset(p_addr, 0, sizeof(*p_addr));
```

Comment:
Style wise I would prefer to see a blank line between these declarations and the first statement.